tests/common/dualtor/dual_tor_utils.py

```diff
@@ -472,3 +473,30 @@ def shutdown(vm_names=None, upper=False, lower=False):
     logger.info('Recover T1 VM ports connected to tor')
     for eos_host, vm_intf in down_intfs:
         eos_host.no_shutdown(vm_intf)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def start_linkmgrd_heartbeat(ptfadapter, duthost, tbinfo):
```

Comment:
1. What `duthost` fixture is this fixture using? Shouldn't this be `duthosts`, and the GARP be sent to both the DUTs in dual-tor testbed?
2. Should this fixture be session scoped instead?

Reply:
The `duthost` fixture is defined in the top level conftest. We don't need to access both `duthosts` objects for the dual ToRs, since we are sending the ARP packets from the PTF through the mux cable which should broadcast them to both ToRs. The `duthosts` fixture here is just used to obtain the DUT/PTF interface mapping which is the same for both DUTs.

I think keeping this as a smaller scope is better, in case the DUT loses the ARP entries during testing (e.g. during `config reload`). It may even be a good idea to make this into a service that runs on the PTF and sends the ARP packets periodically as many of our test cases would cause the ARP entries on the DUT to be lost.

Reply:
1. Now I am thinking will a smaller scope be better? If it is module scope, and a module has more than a few testcases, every testcase is dependent on what the first or previous testcases left (in terms of MACs in DUT neighbor table).
2. Or/and, can we start a thread (in this fixture) that periodically sends GARP packets, no matter what testcase is running?

Reply:
Both good points. I'll update it to function scope for now. I think as per our discussion yesterday, continuously sending GARP packets would be significantly different from what is expected in production. We should instead aim to have some mechanism on the ToR that can re-learn MAC entries after a disruptive event.

Reply:
Got it, thanks! Btw, if you don't specify any scope explicitly, the default scope is `function`.